### PR TITLE
Update datamodel for read/create contributions API

### PR DESCRIPTION
**Pull request by META-DREAMER**
## Motivation and Context

Allows 3rd party integrators / api users to programatically read and create contributions for a given circle.

## Description

Added columns and appropriate permissions in Hasura for the `api-user` role to create and read contributions. Also added a new column to contributions that tracks which API key was used to create the contribution (if any).

## Test and Deployment Plan

Ensure permissions are correctly configured and contributions created with a given API key are marked as such.

## Reviewers

@topocount @zashton 

## Related Issue

#1545

### CODE REVIEW

File: hasura/migrations/default/1667278140823_contributions_api/down.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+alter table "public"."contributions" drop constraint "contributions_created_with_api_key_hash_fkey";
+
+-- Could not auto-generate a down migration.
+-- Please write an appropriate down migration for the SQL below:
```
Comment (crabsinger) [inline hasura/migrations/default/1667278140823_contributions_api/down.sql:5]: can we remove the commented out stuff.

File: hasura/metadata/databases/default/tables/public_vaults.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 table:
-  schema: public
   name: vaults
+  schema: public
```
Comment (crabsinger) [inline hasura/metadata/databases/default/tables/public_vaults.yaml:3]: Seems a bunch of these changes are just re-ordering the fields... would be preferable to keep that change out of here, or do it first in a separate PR.

Comment (META-DREAMER) [reply hasura/metadata/databases/default/tables/public_vaults.yaml]: seems like it was done automatically by Hasura CLI. Related issue: https://github.com/hasura/graphql-engine/issues/8743

will revert in this PR for now

Comment (topocount) [reply hasura/metadata/databases/default/tables/public_vaults.yaml]: @crabsinger I think there are a bunch of hasura files that need to be regenerated since the upgrades

**Review by topocount — APPROVED** [review]
Thanks @META-DREAMER! We should make this step a standard part of augmenting API I suppose.
